crates/ruff_server/src/session/index.rs

```diff
@@ -159,14 +156,7 @@ impl Index {
                 self.notebook_cells
                     .insert(opened_cell.uri.clone(), path.clone());
             }
-            for closed_cell in did_close.iter().flatten() {
```

Comment:
Does that mean that we never retain cells until didClose is called? Does the specification explain why vs code might reference cells after they have been closed ?

Reply:
VS Code uses `didClose` to close a notebook cell. Unfortunately I can't find anything in the specification about this.